services/worker/src/worker/job_runners/dataset/hub_cache.py

```diff
@@ -51,23 +51,29 @@ def compute_hub_cache_response(dataset: str) -> tuple[DatasetHubCacheResponse, f
     viewer = content["viewer"]
     is_valid_progress = is_valid_response["progress"]
 
-    size_response = get_previous_step_or_raise(kind="dataset-size", dataset=dataset)
-    content = size_response["content"]
-    if (
-        "partial" not in content
-        or not isinstance(content["partial"], bool)
-        or "size" not in content
-        or "dataset" not in content["size"]
-        or "num_rows" not in content["size"]["dataset"]
-        or not isinstance(content["size"]["dataset"]["num_rows"], int)
-    ):
-        raise PreviousStepFormatError(
-            "Previous step 'dataset-size' did not return the expected content: 'partial' or 'size.dataset.num_rows'."
-        )
-
-    partial = content["partial"]
-    num_rows = content["size"]["dataset"]["num_rows"]
-    size_progress = size_response["progress"]
+    try:
+        size_response = get_previous_step_or_raise(kind="dataset-size", dataset=dataset)
+        content = size_response["content"]
+        if (
+            "partial" not in content
+            or not isinstance(content["partial"], bool)
+            or "size" not in content
+            or "dataset" not in content["size"]
+            or "num_rows" not in content["size"]["dataset"]
+            or not isinstance(content["size"]["dataset"]["num_rows"], int)
+        ):
+            raise PreviousStepFormatError(
+                "Previous step 'dataset-size' did not return the expected content: 'partial' or 'size.dataset.num_rows'."
+            )
+        partial = content["partial"]
+        num_rows = content["size"]["dataset"]["num_rows"]
+        size_progress = size_response["progress"]
+    except (CachedArtifactNotFoundError, PreviousStepFormatError):
+        raise
+    except Exception:
+        partial = False
+        num_rows = 0
```

Comment:
the hub won't show 0 rows on the dataset page right ?

Reply:
hmmm maybe it should be optional... Looking at it

Reply:
Possible changes soon to this response:
- make num_rows an optional field?
- add num_rows_estimated (see https://github.com/huggingface/dataset-viewer/pull/2906)
- remove "tags"?
- replace "viewer / preview / partial" with one field that would simplify the logic on the Hub side (currently: `!hub_cache.viewer ? "preview" : hub_cache.partial ? "viewer-partial" : "viewer"`) -> `viewer: "viewer" | "viewer-partial" | "preview" | "none"`?

We might as well work all this in a new step (`dataset-moonlanding-cache` for example). And once we've finished this batch of changes, prepare the Hub to consume the new step, and remove `dataset-hub-cache` here.

wdyt @lhoestq?

Reply:
It's currently not used on moonlanding btw. So, we can change it. I agree that `num_rows = 0` is wrong.

Two options:
- `num_rows = None`
- don't set the `num_rows` field

Let's set it to `None`